lib/carrierwave/uploader/processing.rb

```diff
@@ -65,7 +65,11 @@ def process(*args)
           condition_type = new_processors.keys.select { |key| [:if, :unless].include?(key) }[0]
           condition = new_processors.delete(:if) || new_processors.delete(:unless)
           new_processors.each do |processor, processor_args|
-            self.processors += [[processor, processor_args, condition, condition_type]]
+            if processor_args.is_a? Array
+              processor_kwargs, processor_args = processor_args.partition { |processor_arg| processor_arg.is_a? Hash }
+              processor_kwargs = processor_kwargs.reduce(:merge)
```

Comment:
Isn't it simpler to perform this separation inside `#process!`, not here?

Reply:
Yes, you're right. Felt like it would be more readable in the process method but you're right, it's much easier in the `process!` method. Thanks for the review!